### PR TITLE
CES scale test: prelim testing for 5k scale test

### PR DESCRIPTION
**Pull request by thorn3r**
For testing
Swaps out the create-cluster action with a [modified copy on my fork](https://github.com/cilium/scale-tests-action/pull/7) to adjust KCM and scheduler settings to achieve higher scale.

### CODE REVIEW

File: .github/workflows/scale-test-100-gce.yaml [hunk 1]
```diff
@@ -160,14 +161,16 @@ jobs:
...
         with:
           cluster_name: ${{ steps.vars.outputs.cluster_name }}
-          control_plane_size: n1-standard-8
-          control_plane_count: 1
+          control_plane_size: n2-standard-64
+          control_plane_count: 3
+          control_plane_volume_size: 50
           node_size: e2-medium
-          node_count: 100
+          node_count: 20
```
Comment (marseel) [inline .github/workflows/scale-test-100-gce.yaml]: FYI You were trying to run 20 nodes with a test targeted for 100 nodes (`--nodes=100 \` below) so there was not enough room for pods and test timed out.

File: .github/workflows/scale-test-100-gce.yaml [hunk 2]
```diff
@@ -217,6 +220,8 @@ jobs:
           export CL2_PROMETHEUS_SCRAPE_CILIUM_OPERATOR=true
           export CL2_PROMETHEUS_SCRAPE_CILIUM_AGENT=true
           export CL2_PROMETHEUS_MEMORY_SCALE_FACTOR=2.0
+          export CL2_LOAD_TEST_THROUGHPUT=50.0
```
Comment (marseel) [inline .github/workflows/scale-test-100-gce.yaml:334]: Commenting on random line:

Currently, we are running kube-proxy in the cluster. For 5k I think it makes sense to stick with KPR and disabling kube-proxy - services will generate quite a lot of churn. 

Even for 100 nodes it still makes sense to disable it so it can be a totally separate PR.

File: .github/workflows/scale-test-100-gce.yaml [hunk 3]
```diff
@@ -88,7 +88,8 @@ jobs:
             --helm-set=clustermesh.apiserver.image.useDigest=false \
             --helm-set=hubble.relay.image.repository=quay.io/${{ env.QUAY_ORGANIZATION_DEV }}/hubble-relay-ci \
             --helm-set=hubble.relay.image.tag=${SHA} \
-            --helm-set=hubble.relay.image.useDigest=false"
+            --helm-set=hubble.relay.image.useDigest=false \
+            --helm-set=ciliumEndpointSlice.enabled=true"
```
Comment (marseel) [inline .github/workflows/scale-test-100-gce.yaml]: I'm pretty sure we will kill prometheus if we don't disable `endpointHealthChecking.enabled=false` and `healthChecking=false` - just like we recommend in scalability report now. I will create an issue for it to fix it once and for all.  Link: TBD.

File: .github/workflows/scale-test-100-gce.yaml [hunk 4]
```diff
@@ -160,14 +161,16 @@ jobs:
...
           cluster_name: ${{ steps.vars.outputs.cluster_name }}
-          control_plane_size: n1-standard-8
-          control_plane_count: 1
+          control_plane_size: n2-standard-64
+          control_plane_count: 3
+          control_plane_volume_size: 50
           node_size: e2-medium
-          node_count: 100
+          node_count: 20
+          node_volume_size: 50
```
Comment (marseel) [inline .github/workflows/scale-test-100-gce.yaml]: `node_size: e2-standard-8` for heavy-deployment will need to be `e2-standard-32` probably:
https://github.com/kubernetes/test-infra/blob/8c94170825a87c6b6eb339a117263f6a214d0872/config/jobs/kubernetes/sig-scalability/sig-scalability-release-blocking-jobs.yaml#L99C37-L99C51

File: .github/workflows/scale-test-100-gce.yaml [hunk 5]
```diff
@@ -160,14 +161,16 @@ jobs:
...
       - name: Deploy cluster
         id: deploy-cluster
-        uses: cilium/scale-tests-action/create-cluster@238d773bd07754bfd693a6b22c94eddf3a12778d # main
+        uses: thorn3r/scale-tests-action/create-cluster@a9f91acd923775b8b90fa23fb1743ddc8b3efc03  # updates for 5k
         timeout-minutes: 30
         with:
           cluster_name: ${{ steps.vars.outputs.cluster_name }}
-          control_plane_size: n1-standard-8
-          control_plane_count: 1
+          control_plane_size: n2-standard-64
```
Comment (marseel) [inline .github/workflows/scale-test-100-gce.yaml:187]: I just double-checked, and upstream runs  `n2-standard-32`. 
From recent test:
```
gce-scale-cluster-master  us-east1-b  n2-standard-32               10.40.0.3    35.237.108.192  RUNNING
```
https://github.com/kubernetes/test-infra/blob/a4f43f0f14feb4ec582db4c8b4c64ac6ca767b5d/config/jobs/kubernetes/sig-scalability/sig-scalability-presets.yaml#L153

File: .github/workflows/scale-test-100-gce.yaml [hunk 6]
```diff
@@ -217,6 +220,8 @@ jobs:
           export CL2_PROMETHEUS_SCRAPE_CILIUM_OPERATOR=true
           export CL2_PROMETHEUS_SCRAPE_CILIUM_AGENT=true
           export CL2_PROMETHEUS_MEMORY_SCALE_FACTOR=2.0
+          export CL2_LOAD_TEST_THROUGHPUT=50.0
+          export CL2_DELETE_TEST_THROUGHPUT=50.0
```
Comment (marseel) [inline .github/workflows/scale-test-100-gce.yaml:335]: FYI, depending on how much metrics we fetch from cilium-agent, we might want to tune the scrape interval:
https://github.com/kubernetes/perf-tests/pull/2651/files#diff-6a4d7ca416707ad8bf902645ddddde590e8fde0c2711e67ee7e68ea38e593d61R3

I've set it to 30s by default, but I am pretty sure it won't be enough.

File: .github/workflows/scale-test-100-gce.yaml (hunk 1 above)
Comment (thorn3r) [reply .github/workflows/scale-test-100-gce.yaml]: I only wanted to test out the changes I made to reduce the root disk size without wasting so many nodes. Each node was being deployed with 128GB disk which was eating up our quota very quickly (and costing more $)

File: .github/workflows/scale-test-100-gce.yaml (hunk 2 above)
Comment (thorn3r) [reply .github/workflows/scale-test-100-gce.yaml:334]: Good call :+1:

File: .github/workflows/scale-test-100-gce.yaml (hunk 3 above)
Comment (marseel) [reply .github/workflows/scale-test-100-gce.yaml]: FYI, I've started issue with some proposal: https://github.com/cilium/cilium/issues/32820

Comment (thorn3r) [reply .github/workflows/scale-test-100-gce.yaml]: excellent proposal :+1:
